src/network/networkd-routing-policy-rule.c

```diff
@@ -73,8 +75,13 @@ static void routing_policy_rule_hash_func(const RoutingPolicyRule *rule, struct
                 siphash24_compress(&rule->to, FAMILY_ADDRESS_SIZE(rule->family), state);
                 siphash24_compress(&rule->to_prefixlen, sizeof(rule->to_prefixlen), state);
 
+                i = rule->invert_rule;
+                siphash24_compress(&i, sizeof(i), state);
```

Comment:
why not just `siphash24_compress(&rule->invert_rule, sizeof(rule->invert_rule))` here?

Reply:
Is it possible? `rule->invert_rule` is `bool`.

Reply:
should be fine. taking a pointer of a bool via `&` only fails if the bool is a bit field, but this one isn't.

Reply:
For safety(?) I added `siphash24_compress_boolean()`.

Reply:
makes sense!